### ISSUE TITLE
[Feature] allow PinotFSFactory register same scheme with different configurations

### ISSUE THREAD

**Issue by walterddr** [apache/pinot#8474]
Currently there's no way to create/register multiple PinotFS instance for the same scheme.

This is useful when access secrets need to be separately configured for different part of the same file system scheme (for example: S3 with different bucket)

**Comment by KKcorps** [comment]
I can take this up. But deciding which filesystem to pickup won't be possible with just the scheme then. It will require both the scheme as well as the bucket (or similar in case of other FS).

**Comment by walterddr** [comment]
correct. i guess we might need to keep the PinotFSFactory storing not just the scheme as the hash key but scheme + host of the URI.

I haven't thought this through with all the FS that we support so i guess some investigation on 
- is scheme + host enough for our use cases; and is there unnecessary overhead for other FS types (e.g. `file://`?)
- how to fallback to scheme only? (e.g. if one decided to create another pinotFS but reused the same PinotFS instances of the scheme previously defined)
-

**Comment by Jackie-Jiang** [comment]
We can potentially add an optional id under each scheme to differentiate credentials. When retrieving the `PinotFS`, user can pass the optional id. If no id is provided, we keep the existing behavior.
